test/Containers/DenseAxisArray.jl

```diff
@@ -249,6 +249,28 @@ And data, a 0-dimensional $(Array{Int,0}):
         @test (@inferred C[2, [:a, :b]]) == DenseAxisArray([5.0, 6.0], [:a, :b])
         @test (@inferred C[2:3, :b]) == DenseAxisArray([6.0, 8.0], 2:3)
     end
+    @testset "Broadcast" begin
```

Comment:
Could you add `@inferred` in the tests ?

Reply:
Done

Reply:
Although for some reason, `.^ 2` doesn't infer:
```Julia
julia> @code_warntype Base.broadcasted(Base.literal_pow, Main.:^, A, Val(2))
Variables
  #self#::Core.Const(Base.Broadcast.broadcasted)
  f::Core.Const(Base.literal_pow)
  arg1::Core.Const(^)
  arg2::DenseAxisArray{Float64, 2, Tuple{Vector{Symbol}, Vector{Symbol}}, Tuple{JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}, JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}}}
  args::Core.Const((Val{2}(),))
  args′::Tuple{Base.RefValue{Val{2}}}
  arg2′::DenseAxisArray{Float64, 2, Tuple{Vector{Symbol}, Vector{Symbol}}, Tuple{JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}, JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}}}
  arg1′::Base.RefValue{typeof(^)}

Body::DenseAxisArray{Float64, 2, _A, _B} where {_A, _B<:Tuple{JuMP.Containers._AxisLookup, JuMP.Containers._AxisLookup}}
1 ─      nothing
│        (arg1′ = Base.Broadcast.broadcastable(arg1))
│        (arg2′ = Base.Broadcast.broadcastable(arg2))
│        (args′ = Base.Broadcast.map(Base.Broadcast.broadcastable, args))
│   %5 = Core.tuple(arg1′, arg2′)::Tuple{Base.RefValue{typeof(^)}, DenseAxisArray{Float64, 2, Tuple{Vector{Symbol}, Vector{Symbol}}, Tuple{JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}, JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}}}}
│   %6 = Core._apply_iterate(Base.iterate, Base.Broadcast.combine_styles, %5, args′)::Core.Const(JuMP.Containers._DenseAxisArrayBroadcastStyle())
│   %7 = arg1′::Base.RefValue{typeof(^)}
│   %8 = Core.tuple(%6, f, %7, arg2′)::Core.PartialStruct(Tuple{JuMP.Containers._DenseAxisArrayBroadcastStyle, typeof(Base.literal_pow), Base.RefValue{typeof(^)}, DenseAxisArray{Float64, 2, Tuple{Vector{Symbol}, Vector{Symbol}}, Tuple{JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}, JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}}}}, Any[Core.Const(JuMP.Containers._DenseAxisArrayBroadcastStyle()), Core.Const(Base.literal_pow), Base.RefValue{typeof(^)}, DenseAxisArray{Float64, 2, Tuple{Vector{Symbol}, Vector{Symbol}}, Tuple{JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}, JuMP.Containers._AxisLookup{Dict{Symbol, Int64}}}}])
│   %9 = Core._apply_iterate(Base.iterate, Base.Broadcast.broadcasted, %8, args′)::DenseAxisArray{Float64, 2, _A, _B} where {_A, _B<:Tuple{JuMP.Containers._AxisLookup, JuMP.Containers._AxisLookup}}
└──      return %9
```
but everything else does, so I just removed that for now.

Reply:
Is this a regression or it didn't work before either?

Reply:
It was a regression. But I've now fixed it. The compiler didn't like the tuple recursion passing along an additional state argument, so I've split into two passes: first to collect the axes, and then a loop to check they are similar if there is more than one DenseAxisArray in the result.